WcaOnRails/app/views/competitions/index.html.erb

```diff
@@ -1,17 +1,181 @@
+
+        <li class="same-line last-one"><a href="#comp-query-form" class="js-scroll-to">Top</a></li>
+      </ul>
     <% end %>
-  </table>
+  </div>
+
+  <%= form_tag competitions_path, method: :get, class: "competition-select form-inline", id: "comp-query-form" do %>
+    <%= render 'index_form' %>
+  <% end %>
+
+  <% if @competitions.length == 0 %>
+    <div class="alert alert-warning">No competitions found.</div>
+  <% else %>
+    <% if params[:display] == "List" %>
+      <div class="row competition-list">          
+        <% if @in_progress_competitions.length > 0 %>
+          <div class="col-md-12" id="current">  
+            <%= render 'index_table', competitions: @in_progress_competitions, title: "Competitions in progress" %>
+          </div>
+        <% end %>
+
+        <% if @upcoming_competitions.length > 0 %>
+          <div class="col-md-12" id="upcoming">
+            <%= render 'index_table', competitions: @upcoming_competitions, title: "Upcoming competitions" %>
+          </div>
+        <% end %>
+
+        <% if @past_competitions.length > 0 %>
+          <div class="col-md-12" id="past">
+            <%= render 'index_table', competitions: @past_competitions, title: "Past competitions (last 90 days)" %>
+          </div>
+        <% end %>
+      </div>
+    <% else %>
+      <div class="row competition-list">
+        <div class="col-xs-12 col-md-12">
+          <div id="comps-map"></div>
+        </div>
+      </div>
+    <% end %>
+  <% end %>
 </div>
+
+<script>
+// See https://github.com/cubing/worldcubeassociation.org/issues/356
+function mapSize() {
+  var nav = parseInt($(".navbar").outerHeight(true));
+  var head = parseInt($("h2").outerHeight(true));
+  var form = parseInt($(".form-group").outerHeight(true));
+  var foot = parseInt($(".footer").outerHeight(true));
+
+  var height = parseInt($(window).innerHeight());
+
+  var map = height - nav - head - form - foot;
+
+  $("#comps-map").height( map );
+}
+
+function affixWidth() {
+  // From https://github.com/twbs/bootstrap/issues/6350#issuecomment-16069663
+  // also http://stackoverflow.com/a/20301984
+  // and http://stackoverflow.com/a/24908611
+  var affixBase = $(".container");
+  var width = affixBase.width() - parseInt(affixBase.css('padding-left')) - parseInt(affixBase.css('padding-right'));
+  $("#simple-nav").width(width);
+}
+
+// From http://stackoverflow.com/a/30966965
+$(".js-scroll-to").click(function(e) {
+  e.preventDefault();
+
+  // Only apply the movement if there is a div to go to.
+  if(!$(this).hasClass('disabled')) {
+    // Little hack to activate the affix thing and
+    // move correctly to the anchor.
+    $("#simple-nav").removeClass('affix-top').addClass('affix');
+
+    var destination = $(this).attr('href');
+
+    $('html, body').animate({
+      scrollTop: $(destination).offset().top - 40
```

Comment:
There must be a bootstrap variable for our navbar height. If not, couldn't
you compute the height using javascript? Bootstrap variable is preferable
for performance, I would think.

On Wed, Mar 23, 2016, 05:34 Pedro Santos <notifications@github.com> wrote:

> In WcaOnRails/app/views/competitions/index.html.erb
> <https://github.com/cubing/worldcubeassociation.org/pull/319#discussion_r57151280>
> :
>
> > +}
> > +
> > +// From http://stackoverflow.com/a/30966965
> > +$(".js-scroll-to").click(function(e) {
> > +  e.preventDefault();
> > +
> > +  // Only apply the movement if there is a div to go to.
> > +  if(!$(this).hasClass('disabled')) {
> > +    // Little hack to activate the affix thing and
> > +    // move correctly to the anchor.
> > +    $("#simple-nav").removeClass('affix-top').addClass('affix');
> > +
> > +    var destination = $(this).attr('href');
> > +
> > +    $('html, body').animate({
> > +      scrollTop: $(destination).offset().top - 40
>
> It's the navbar height.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/cubing/worldcubeassociation.org/pull/319/files/6a2cfebb820da1014ae577d72cfa04d86d7e34db#r57151280>
>